tests/test_generator_iter.py

```diff
@@ -0,0 +1,85 @@
+import subprocess as sb
+
+
+def run(command):
+    return sb.run(command.split(), capture_output=True) \
+               .stdout \
+               .decode() \
+               .strip()
+
+def hash_password(password, salt="hi"):
+    return run(f"./encr -p {password} -s {salt}")
+
+
+def base_call_found(password, alphabet="abc"):
+    length = len(password)
+    hashed = hash_password(password, "hi")
+    result = run(f"./brute -g -i -l {length} -h {hashed} -a {alphabet}")
+    assert result == f"Password found: '{password}'"
+
+
+def base_call_notfound(password, alphabet="abc"):
+    length = len(password)
+    hashed = hash_password(password, "hi")
+    result = run(f"./brute -g -i -l {length} -h {hashed} -a {alphabet}")
+    assert result == f"Password not found"
+
+# Length 1
```

Comment:
Standard practice is to have:
1. One happy path test: "abc", not found
2. Tests for all corner cases: "aaa", "ccc"
3. Performance test: adaptive test that will find password length for which run takes some measurable time (>0.1s) and will check that for length+1 consumed time grew linearly (on a length of alphabet).